### PR TITLE
Add descriptive comments to `ConversationMemberUpdate`

### PR DESCRIPTION
**Pull request by mdimjasevic**

### CODE REVIEW

**Review by jschaul — APPROVED** [review]
Perfect!

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs [hunk 1]
```diff
@@ -74,8 +74,18 @@ data ConversationMemberUpdate = ConversationMemberUpdate
   { cmuTime :: UTCTime,
     cmuOrigUserId :: Qualified UserId,
     cmuConvId :: Qualified ConvId,
-    cmuAlreadyPresentUsers :: [UserId], -- pre-existing users in the conversation from the receiving domain
+    -- | This field notifies a remote backend about which users need to be sent
```
Comment (pcapriotti) [inline libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:77]: A field doesn't "notify". Maybe just write something like "users from this backend who need to ..."

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs [hunk 2]
```diff
@@ -74,8 +74,18 @@ data ConversationMemberUpdate = ConversationMemberUpdate
...
+    -- notifications about this change. This is required as we do not expect a
+    -- non-conversation owning backend to have a indexed mapping of conversation
```
Comment (pcapriotti) [inline libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:79]: Typo: "an indexed"

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs [hunk 3]
```diff
@@ -74,8 +74,18 @@ data ConversationMemberUpdate = ConversationMemberUpdate
...
+    -- to users.
+    cmuAlreadyPresentUsers :: [UserId],
+    -- | Users which got added to the conversation.
```
Comment (pcapriotti) [inline libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:82]: Either s/which/who or s/which got //

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs [hunk 4]
```diff
@@ -74,8 +74,18 @@ data ConversationMemberUpdate = ConversationMemberUpdate
...
     cmuUsersAdd :: [(Qualified UserId, RoleName)],
+    -- | Users which got removed from the conversation. This should probably be
```
Comment (pcapriotti) [inline libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:84]: Ditto

**Review by pcapriotti — APPROVED** [review]
Looks good. Minor nitpicks follow.

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs (hunk 2 above)
Comment (mdimjasevic) [reply libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:79]: Thanks! Addressing that...

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs (hunk 1 above)
Comment (mdimjasevic) [reply libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:77]: Thanks! Incorporating that...

File: libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs (hunk 3 above)
Comment (mdimjasevic) [reply libs/wire-api-federation/src/Wire/API/Federation/API/Galley.hs:82]: Ok, I'll do a variation of it.

**Pull request merged**